### DISCUSSION TITLE
Use a connection string from KeyVault in Development

### DISCUSSION THREAD

**Discussion by UrielMhezzek** [microsoft/aspire#1568] [Q&A]
I want to add a MS SQL database to my Blazor Web App. The connection string comes from Key Vault and I want to migrate it later. In dev mode I would like to use a localdb connection string as an alternative. I addet all components like .net 7 but it dosn't work. Aspire wants to orchester the Database. How can i do it?

**Comment by josephaw1022** [comment]
Here is a good reference

 fohttps://github.com/josephaw1022/ScaleStore/blob/master/KEDAScalingUI.AppHost/Program.csr what I think you are getting at. 



can you provide an example repo to better illustrate what exactly the problem is

**Reply by UrielMhezzek** [reply]
see my comment

**Comment by UrielMhezzek** [comment]
You can download my test project. I have removed most things to reproduce the error. 

https://github.com/UrielMhezzek/aspire_example/tree/feature/adding_database

Actually i get an error in Tik.Backend.DatabaseExtentions.InitializeDatabase
at  var context = scope.ServiceProvider.GetRequiredService<ApplicationDBContext>();

System.InvalidOperationException: 'ConnectionString is missing. It should be provided in 'ConnectionStrings:Server=(localdb)\mssqllocaldb;Database=test-Debug;Trusted_Connection=True;MultipleActiveResultSets=true;AttachDbFileName = D:\OneDrive\LocalDB\Debug\test-Debug.mdf;' or under the 'ConnectionString' key in 'Aspire:Microsoft:EntityFrameworkCore:SqlServer' or 'Aspire:Microsoft:EntityFrameworkCore:SqlServer:ApplicationDBContext' configuration section.'

**Reply by UrielMhezzek** [reply]
I can Add this

```
"Aspire": {
  "Microsoft": {
    "EntityFrameworkCore": {
      "SqlServer": {
        "ApplicationDBContext": {
          "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=test-Debug;Trusted_Connection=True;MultipleActiveResultSets=true;AttachDbFileName = D:\\OneDrive\\LocalDB\\Debug\\test-Debug.mdf;",
          "DbContextPooling": true,
          "HealthChecks": false,
          "Tracing": false,
          "Metrics": true
        }
      }
    }
  }
}
```

to my Backend Appsettings. So i get "Only the invariant culture is supported in globalization-invariant mode" error. 

This is my next Pitfall, i think. 
1.: I think, that should not to be so.
2.: I think, aspirehost should orchester it and get the connectionstring from secure key vault and not from unsecure appsettings
3.: I real scenario i don't know something about the connection string. Only the Key Vault Manager knows about the string.

**Comment by davidfowl** [comment]
I updated the discussion title to match what the discussion is about. It seems like the request here is the ability to reference a connection string from key vault secret, is that correct? 

Then there's the discussion about changing the behavior in local dev (using local db or a container vs deployment where maybe you do want to use keyvault)? 

We haven't built a solid end to end here, but I'm hoping we can land on something clean in the next preview.

cc @mitchdenny

**Reply by UrielMhezzek** [reply]
You are right. I would like to create the following. 
First: I want to read the database connection string from Key Vault.
Second: The database in local/dev is a localdb. This database should not be a container.
Third: Later, in Stage and Production, the database should be an Azure Serverless MS SQL Database.
In my environment, this is a developer standard.
